src/coreclr/tools/aot/crossgen2/Program.cs

```diff
@@ -499,6 +499,19 @@ private int Run(string[] args)
 
                     RunSingleCompilation(singleCompilationInputFilePaths, instructionSetSupport, compositeRootPath, unrootedInputFilePaths, singleCompilationVersionBubbleModulesHash, typeSystemContext);
                 }
+
+                // In case of inputbubble ni.dll are created as ni.dll.tmp in order to not interfere with crossgen2, move them all to ni.dll
+                // See https://github.com/dotnet/runtime/issues/55663#issuecomment-898161751 for more details
+                if (_commandLineOptions.InputBubble)
```

Comment:
This needs to be conditional on `OutNearInput` as well.

Reply:
Thanks. I think I'll add check and throw somewhere above that `SingleFileCompilation` uses `OutNearInput`, because it makes no sense without it